### PR TITLE
Adds internal Compiler.CommonWithMetadataImportOptions and removes SymUnmanagedExtensions

### PR DESCRIPTION
**Pull request by tmat**
**Customer scenario**

No observable change. This is a prerequisite of a coming bigger change in PDB testing infrastructure.

**Bugs this fixes:**

**Workarounds, if any**

**Risk**

Small.

**Performance impact**

none.

**Is this a regression from a previous update?**

**Root cause analysis:**

**How was the bug found?**

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov @dotnet/roslyn-compiler Trivial change. Please review.

**Comment by ivanbasov** [comment]
:shipit:

**Comment by tmat** [comment]
@dotnet/roslyn-compiler  Ping

File: src/Compilers/Core/Portable/Compilation/CompilationOptions.cs [hunk 1]
```diff
@@ -525,6 +527,7 @@ public CompilationOptions WithOverflowChecks(bool checkOverflow)
         protected abstract CompilationOptions CommonWithCryptoPublicKey(ImmutableArray<byte> cryptoPublicKey);
         protected abstract CompilationOptions CommonWithDelaySign(bool? delaySign);
         protected abstract CompilationOptions CommonWithCheckOverflow(bool checkOverflow);
+        internal abstract CompilationOptions CommonWithMetadataImportOptions(MetadataImportOptions value);
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: Why `internal` here when everything else is `protected`?

Comment (tmat) [inline src/Compilers/Core/Portable/Compilation/CompilationOptions.cs]: MetadataImportOptions is internal. Protected would expose it outside of the assembly.

**Comment by tmat** [comment]
@MattGertz For ask mode approval. Is this ok to merge into master or should I rebase to 15.6 branch? This change is simple and will allow us to do add more PDB testing in a follow up PR that has test only changes.

**Comment by cston** [comment]
LGTM

**Comment by MattGertz** [comment]
Approved for master.

**Pull request merged**